mcstasscript/interface/instr.py

```diff
@@ -1989,19 +1992,51 @@ def backengine(self):
         # Load data and store in __data
         #data = simulation.load_results()
         #self._set_data(data)
+        
+        ## look for MCPL_output components and the defined filenames
+        self.add_mcpl_to_output(simulation)
 
+        # simulation results from .dat files loaded as dict
         data = simulation.load_results()
         data_dict = {"data": data}
-        key = self.output_keys[0]
-        output_data = self.output[key]
+        # adding to the libpyvinyl output datacollection with key = sim_data_key
+        sim_data_key = self.output_keys[0]
+        output_data = self.output[sim_data_key] 
         output_data.set_dict(data_dict)
 
-        if "data" not in self.output.get_data():
+        if "data" not in self.output[sim_data_key].get_data():
             print("\n\nNo data returned.")
             return None
         else:
-            return self.output.get_data()["data"]
+            return self.output[sim_data_key].get_data()["data"]
+
+    def add_input_to_mcpl(self):
+        try:
+            mcpl_file = self.input["mcpl"].filename
+            for comp in self.component_list:  # starting from the last one!
+                if comp.component_name == "MCPL_input":
+                    comp.filename = '"' + mcpl_file + '"'
+                    break
+        except:
+            return
```

Comment:
Changing the filename to include the extra quotes in this way is a bit dangerous as you are making the assumption that there is only the one MCPL_input component, you can insert several if you wish (though its a rare use-case)

Reply:
One could add a check if multiple components with name "MCPL_input" exist and raise an error. In case of multiple MCPL_input the user should not use the input but set the names by hand.